binary-decode.go

```diff
@@ -81,7 +94,25 @@ func (cdc *Codec) _decodeReflectBinary(bz []byte, info *TypeInfo, rv reflect.Val
 		rv = rv.Elem()
 	}
 
-	var _n int
+	// Handle override if a pointer to rv implements UnmarshalWire.
+	if info.IsWireUnmarshaler {
+		// First, decode repr instance from bytes.
```

Comment:
So why don't we just invoke UnmarshalWire directly and let the method handle all this?

The second call to _decodeReflectBinary on line 105 worries me about endless recursion.

Reply:
mmm then it would be less type-safe because MarshalWire/UnmarshalWire would have to specify the repr-type as only interface{} or equally generic.  As specified, due to the flexibility of the method signature, we cannot make this an interface... but it's not really a problem for either reflection or code generation.